### ISSUE TITLE
[BUG] There is no projects in my console UI

### ISSUE THREAD

**Issue by StreamSun** [flyteorg/flyte#6037]
### Describe the bug

<img width="954" alt="Screenshot 2024-11-20 143841" src="https://github.com/user-attachments/assets/72e2a1c1-c17a-4a89-a3c9-1fb55bbe5409">

### Expected behavior

There is no projects in my console UI when I deploy it use flyte-core, but if I use flyte-binary to show the projects, it works

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by punkerpunker** [comment]
I am experiencing exact same issue. `flyte-binary` works, `flyte-core` - emtpy.

**Comment by davidmirror-ops** [comment]
@StreamSun @punkerpunker what flyte and flytekit versions are you using? Is this flyte-core deployment behind a proxy?
There's the `initialProjects` setting which is particular to `flyte-core`, please check if it's there in your config.

**Comment by punkerpunker** [comment]
Hey @davidmirror-ops, thanks for the quick reply. I've deployed `flyte-core` following your [guide](https://github.com/davidmirror-ops/flyte-the-hard-way/blob/main/docs/on-premises/single-node/002-single-node-onprem-install.md), using helm chart with the following version:

```
- name: flyte-core
  version: 1.14.1
  repository: https://flyteorg.github.io/flyte
```

If that helps - I've also checked postgres - project table isn't empty:
```
root@postgres-8b6895f5c-cdpjd:/# psql -U flyte -d flyte
psql (13.5 (Ubuntu 13.5-0ubuntu0.21.04.1))
Type "help" for help.

flyte=# select * from projects;
 id |          created_at           |          updated_at           | deleted_at |  identifier   |     name      |        description        |                   labels                   | state 
----+-------------------------------+-------------------------------+------------+---------------+---------------+---------------------------+--------------------------------------------+-------
  1 | 2025-01-09 16:50:42.739772+00 | 2025-01-09 16:50:42.739772+00 |            | flytesnacks   | flytesnacks   | flytesnacks description   |                                            |     0
  2 | 2025-01-09 16:50:42.743719+00 | 2025-01-09 16:50:42.743719+00 |            | flytetester   | flytetester   | flytetester description   |                                            |     0
  3 | 2025-01-09 16:50:42.746655+00 | 2025-01-09 16:50:42.746655+00 |            | flyteexamples | flyteexamples | flyteexamples description |                                            |     0
  4 | 2025-01-09 17:15:28.266146+00 | 2025-01-09 17:15:28.266146+00 |            | gleb          | gleb          | gleb                      | \x0a04676c65621204676c65622204676c65622a00 |     0
(4 rows)
```

However, any call to UI returns `304` and keeps showing empty interface, even if following the direct link to the run given by `pyflyte`. 

If that helps - I can try running another version of `flyte-core`.

Thank you!

**Comment by punkerpunker** [comment]
@davidmirror-ops 
As frontend is communicating to the FlyteAdmin with the ADMIN_API_URL provided by configmap, we need to modify it for the sandbox installation so it communicates to it via port-forwarded loopback.

To avoid CORS issues, we also need to modify adminServer allowedOrigins.

See the following PR: https://github.com/davidmirror-ops/flyte-the-hard-way/pull/24

**Comment by davidmirror-ops** [comment]
Yeah, this is very likely a CORS issue. See https://github.com/flyteorg/flyte/pull/5855

**Comment by davidmirror-ops** [comment]
@punkerpunker @StreamSun Flyte 1.14 should handle this better (https://github.com/flyteorg/flyte/pull/5855) Let us know otherwise
